addons/mail/controllers/im_status.py

```diff
@@ -0,0 +1,30 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import http
+from odoo.http import request
+
+from datetime import datetime, timedelta, time
+
+
+class ImStatusController(http.Controller):
+    @http.route("/mail/im_status", methods=["POST"], type="json", auth="user")
+    def set_im_status(self, action):
+        user = request.env.user
+        user.forced_im_status = None if action == "online" else action
```

Comment:
I'm not sure what you mean.
By not sending anything how will the server know to reset the `forced_im_status`?

Reply:
It just feel odds that setting the status to "online" actually *clears* the field instead of assigning said value. 
Maybe something like this?

```js
rpc("/mail/im_status", { action: null/false });
```